### PR TITLE
Add GetVault Api, some minor updates

### PR DESCRIPTION
**Pull request by zhmolly11**

### CODE REVIEW

File: internal/types/keysign.go [hunk 1]
```diff
@@ -6,48 +6,61 @@ import (
...
 
-type KeysignRequest struct {
-	PublicKeyECDSA   string   `json:"public_key_ecdsa"`   // ECDSA public key, used to identify the backup file
-	Messages         []string `json:"messages"`           // Messages need to be signed
-	Session          string   `json:"session"`            // Session ID , it should be an UUID
-	HexEncryptionKey string   `json:"hex_encryption_key"` // Hex encryption key, used to encrypt the keysign messages
-	DerivePath       string   `json:"derive_path"`        // Derive Path
-	IsECDSA          bool     `json:"is_ecdsa"`           // indicate use ECDSA or EDDSA key to sign the messages
-	VaultPassword    string   `json:"vault_password"`     // password used to decrypt the vault file
+type Coin struct {
```
Comment (johnnyluo) [inline internal/types/keysign.go:13]: Let's not redefine the structure here , just use the proto in `commondata`

File: api/server.go [hunk 2]
```diff
@@ -291,12 +299,48 @@ func (s *Server) SignMessages(c echo.Context) error {
...
+	}
+
+	serializedData, err := proto.Marshal(keysignMsg)
+	if err != nil {
+		return fmt.Errorf("fail to Marshal keygenMsg, err: %w", err)
+	}
+
+	var compressedData bytes.Buffer
+	// Create a new XZ writer.
+	xzWriter, err := xz.NewWriter(&compressedData)
```
Comment (johnnyluo) [inline api/server.go:326]: can you please move this compress/decompress logic into a dedicated func

File: internal/types/keysign.go [hunk 3]
```diff
@@ -6,48 +6,43 @@ import (
...
-	HexEncryptionKey string   `json:"hex_encryption_key"` // Hex encryption key, used to encrypt the keysign messages
-	DerivePath       string   `json:"derive_path"`        // Derive Path
-	IsECDSA          bool     `json:"is_ecdsa"`           // indicate use ECDSA or EDDSA key to sign the messages
-	VaultPassword    string   `json:"vault_password"`     // password used to decrypt the vault file
+type KeysignAPIRequest struct {
+	PublicKey  string                       `json:"public_key"`  // public key, used to identify the backup file
+	Messages   []string                     `json:"messages"`    // Messages need to be signed
+	DerivePath string                       `json:"derive_path"` // Derive Path
+	IsECDSA    bool                         `json:"is_ecdsa"`    // indicate use ECDSA or EDDSA key to sign the messages
+	Payload    *keysignTypes.KeysignPayload `json:"payload"`     // keysign payload
```
Comment (johnnyluo) [inline internal/types/keysign.go:19]: This is a misunderstanding , `Payload` should not exist in here ,  client side UI should construct the transaction , and then get the PreKeysignImage, the bytes get from wallet-core ,  and put it into `Messages` field 

No point of passing `Payload` to server side , as server will not able to parse it , and won't be able to construct the transaction either.

File: api/server.go [hunk 4]
```diff
@@ -263,6 +264,43 @@ func (s *Server) DownloadVault(c echo.Context) error {
...
+	if err != nil {
+		return fmt.Errorf("fail to decrypt vault from the backup, err: %w", err)
+	}
+
+	return c.JSON(http.StatusOK, types.VaultGetResponse{
+		Name:           vault.Name,
+		PublicKeyEcdsa: vault.PublicKeyEcdsa,
+		PublicKeyEddsa: vault.PublicKeyEddsa,
+		HexChainCode:   vault.HexChainCode,
+		KeyShares:      vault.KeyShares,
```
Comment (johnnyluo) [inline api/server.go:299]: can you please not return "keyshares" , because client side doesn't need keyshares

Comment (0xkobe) [reply api/server.go]: done

**Review by johnnyluo — APPROVED** [review]
LGTM

**Pull request merged**